### PR TITLE
Allowed to exclude unwanted demo content collections.

### PR DESCRIPTION
**Pull request by sonnykt**
## Changes
1. Introduce a new hook allowing other modules (and consumer sites) to exclude unwanted demo collections.
2. Refactored some code for Drupal 9 compatibility.

On-behalf-of: @salsadigitalauorg <sonny@salsadigital.com.au>

### CODE REVIEW

File: src/Plugin/yaml_content/process/ProcessFile.php [hunk 1]
```diff
@@ -21,21 +22,21 @@ class ProcessFile extends File {
...
     if ($output !== FALSE) {
       $destination = 'public://';
-      // Look-up the field's directory configuation.
-      if ($directory = $context->getField()->getSetting('file_directory')) {
+      // Look-up the field's directory configuration.
+      $directory = $context->getField()->getSetting('file_directory');
+      if ($directory) {
         $directory = trim($directory, '/');
         $directory = PlainTextOutput::renderFromHtml('tide_demo_content');
         if ($directory) {
```
Comment (sonnykt) [inline src/Plugin/yaml_content/process/ProcessFile.php:42]: @MdNadimHossain you may want to check these lines as they don't make sense to me. Not sure what was your intention here hence I couldn't fix the logic.

**Comment by anthony-malkoun** [comment]
@sonnykt what is the use case for this? Why do sites with demo content care about what other modules install demo content?

**Comment by sonnykt** [comment]
> @sonnykt what is the use case for this? Why do sites with demo content care about what other modules install demo content?

@anthony-malkoun it's mainly used in content-health now, but I think it will make more sense for non-DPC sites to exclude the default collection.  Consumer sites may add more fields to the default content types but the default demo content doesn't have data for those fields. The default demo collections become useless for those sites as they have to add their own set of demo content anyway

**Review by vincent-gao — APPROVED** [review]
looks good to me !

**Pull request merged**